packages/Pagination/src/Pagination.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import Input from '@blaze-react/input';
+import usePagination from './hooks/usePagination';
+
+interface PaginationProps {
+  totalItems: number;
+  currentPage: number;
+  visiblePages: number;
+  itemsPerPage?: number;
+  onPageChange: ({ pageNumber, itemsPerPage, offset }: { pageNumber: number, itemsPerPage: number, offset: number }) => void;
+}
+
+const Pagination: React.FC<PaginationProps> = ({ totalItems, currentPage, onPageChange, visiblePages, itemsPerPage }) => {
+
+  const { page, handlePageChange, calculatePages, handleOnItemsPerPage } = usePagination({
+    totalItems,
+    currentPage,
+    visiblePages,
+    onPageChange,
+    defaultItemsPerPage: itemsPerPage
+  });
+
+  const getItemClassName = (number: number) =>
+    `pagination__item ${number === page ? "pagination__item--active" : ""
+    }`;
+
+  const pages = calculatePages();
+
+  return (
+    <div className="pagination">
+      {itemsPerPage && <div className="pagination__rows">
+        <span>Displaying</span>
```

Comment:
Do we wante `Displaying` and `rows per page` labels to be static? maybe they could be pass as props and default to what is showing now